rp2040-hal/src/pio.rs

```diff
@@ -754,6 +754,66 @@ impl<SM: ValidStateMachine> Tx<SM> {
         unsafe { &*self.block }
     }
 
+    /// Gets the FIFO's DMA address
+    pub fn dma_address(&self) -> u32 {
```

Comment:
Why call it `dma_address`? DMA is just your use case, not a property of the returned value. I agree that the function is probably mainly useful for DMA, but it can still be confusing if somebody expects that this address is somehow special.
What about `pub fn fifo_ptr(&self)`?
And I'd return a `*mut u32`, even if that means that in the DMA use case, the caller has to do the `as u32` conversion.